### ISSUE TITLE
sam_FineTune.py

### ISSUE THREAD

**Issue by leehyeonbeen** [hyeonbeenlee/segment-anything-fine-tuning#1]
# Existing issues

1. Batch size more than 1 cause error (due to multi-prompt)  
https://github.com/facebookresearch/segment-anything/issues/277
Temporarily using unit-sized batch gradient accumulation

2. Layerwise LR decay of 0.8 not properly implemented
4. Drop-path with rate of 0.4 not implemented
5. Loss function(Focal+Dice) is not strictly validated, but the training looks fine in running.
6. Decreasing LR with factor of 10 at iteration 60000, 86666...not considered
7. Multiprocessing image loading not properly working

**Comment by EnKaiZhang** [comment]
Why is there no mask in predict after fine tune?

**Comment by FeifanFu** [comment]
> # Existing issues
> 1. Batch size more than 1 cause error (due to multi-prompt)
>    [RuntimeError when using batch size > 1 facebookresearch/segment-anything#277](https://github.com/facebookresearch/segment-anything/issues/277)
>    Temporarily using unit-sized batch gradient accumulation
> 2. Layerwise LR decay of 0.8 not properly implemented
> 3. Drop-path with rate of 0.4 not implemented
> 4. Loss function(Focal+Dice) is not strictly validated, but the training looks fine in running.
> 5. Decreasing LR with factor of 10 at epoch 60000, 86666...not considered
> 6. Multiprocessing image loading not properly working
Is it possible to use the automatic_mask_generator function to make predictions after fine tune?

**Comment by FeifanFu** [comment]
Is it possible to use the automatic_mask_generator function to make predictions after fine tune?

**Comment by leehyeonbeen** [comment]
> Is it possible to use the automatic_mask_generator function to make predictions after fine tune?

The trained model will save ```segment-anything.modeling.sam.Sam.mask_decoder.state_dict()``` to ```models/*.pt```. It will work fine if you properly load fine-tuned mask decoder params to the original pretrained model instance.

**Comment by FeifanFu** [comment]
> > Is it possible to use the automatic_mask_generator function to make predictions after fine tune?
> 
> The trained model will save `segment-anything.modeling.sam.Sam.mask_decoder.state_dict()` to `models/*.pt`. It will work fine if you properly load fine-tuned mask decoder params to the original pretrained model instance.

Thank you for your answer, I tried to train with reference to your project code and got the model after fine tuning, but trying to segment the image in . /tmp folder of sam_firsttry.ipynb for image segmentation, I get an empty list of mask, while the model without fine-tuning can get mask.Does your trained model work properly?

**Comment by leehyeonbeen** [comment]
> > > Is it possible to use the automatic_mask_generator function to make predictions after fine tune?
> > 
> > 
> > The trained model will save `segment-anything.modeling.sam.Sam.mask_decoder.state_dict()` to `models/*.pt`. It will work fine if you properly load fine-tuned mask decoder params to the original pretrained model instance.
> 
> Thank you for your answer, I tried to train with reference to your project code and got the model after fine tuning, but trying to segment the image in . /tmp folder of sam_firsttry.ipynb for image segmentation, I get an empty list of mask, while the model without fine-tuning can get mask.Does your trained model work properly?

I don't recommend using that notebook. It's just for my practice. I used ```visualize.plot_mask``` to actually obtain masks and draw images in the readme file.

**Comment by FeifanFu** [comment]
I understand. Thank you for your reply
